src/misc/CustomAvatar/CustomAvatar.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Box, Typography, makeStyles } from '@material-ui/core';
+import {
+  red,
+  pink,
+  purple,
+  deepPurple,
+  indigo,
+  blue,
+  lightBlue,
+  cyan,
+  teal,
+  green,
+  lightGreen,
+  lime,
+  yellow,
+  amber,
+  orange,
+  deepOrange,
+} from '@material-ui/core/colors/';
+
+const COLORS = [
+  red,
+  pink,
+  purple,
+  deepPurple,
+  indigo,
+  blue,
+  lightBlue,
+  cyan,
+  teal,
+  green,
+  lightGreen,
+  lime,
+  yellow,
+  amber,
+  orange,
+  deepOrange,
+];
+
+function getHash(s) {
+  if (s === undefined || s === null || s.length === 0) return 0;
+  let hash = 0;
+  let i;
+  let chr;
+  for (i = 0; i < s.length; ++i) {
+    chr = s.charCodeAt(i);
+    hash = (hash << 5) - hash + chr;
+    hash |= 0;
+  }
+  return Math.abs(hash);
+}
+
+const getColorFromInt = (h) => {
+  const hash = (h + 64) % 128;
+
+  const col = hash % 16;
+  const colorName = COLORS[col];
+
+  const row = (hash - col) / 16;
+  return colorName[(row + 1) * 100];
+};
+
+const useStyles = makeStyles((theme) => ({
+  avatar: {
+    borderRadius: '50%',
+    width: '24px',
+    backgroundColor: (props) => props.avatarBackgroundColor,
+    color: (props) => theme.palette.getContrastText(props.avatarBackgroundColor),
+  },
+}));
+
+export const CustomAvatar = (props) => {
+  const { userName } = props;
```

Comment:
To make the code slightly more compact, we could replace these 2 lines by a single line:
`export const CustomAvatar = ({userName}) => {`